### DISCUSSION TITLE
Delta DTB 4824 and Artisan.

### DISCUSSION THREAD

**Discussion by finermax1** [artisan-roaster-scope/artisan#1291] [Q&A]
Hello:
I have a question about how to connect the Pid with Artisan. I had a Delta DTB 4824 Pid and Ifd6500. 
After installing the driver, I found that the Pid didn't work with Artisan. The temperature doesn't show up or just -1. So I am getting a Rs 485 to Usb connector. Can I get some advices to solve the problem? Thanks

**Comment by Terracotta-6** [comment]
https://github.com/artisan-roaster-scope/artisan/discussions/716

**Reply by Terracotta-6** [reply]
Delta DTB 4824   Register 4096   . SV 4097 .  P 4105 .  I 4106  . d 4107 .

**Comment by finermax1** [comment]
Thanks, but I tried those numbers with no luck. Should I connect the ifd6500 to the Pid? I am not sure I have put those numbers in the right place.

**Comment by finermax1** [comment]
Hello:
I asked many people.No one could tell me how to do. What shouLd I do?

Thanks

L

**Reply by bigl00z3** [reply]
Hello, 
I haven't used the [Delta DTB](https://deltaacdrives.com/Delta-DTB-User-Manual.pdf) nor the [Ifd6500](https://www.deltaww.com/en-us/products/Industrial-Fieldbus-Solution/52/) but on my side I have a [Fuji PXF](https://www.fujielectric.com/products/instruments/products/controller/PXF.html) and a [USconverter](https://www.usconverters.com/usb-rs485-converter-xs885) converter (I have also used successfully the cheap [JoyIT](https://joy-it.net/en/products/SBC-TTL-RS485) converter).

I can think of a few point you have to be carefull when setting up:

- Does your Ifd6500 need drivers? Are they propery installed? Both the USconverter and the JoyIT generally do  not need additional drivers due to the widely used chip thay are based on.

Once you are sure that your computer "sees" your converteryou can check the following:

When looking at the screenshots that @Terracotta-6 shared in #716 the registers are likely the same but depending on how you have configured your DTB, you may neeed to change the following:
- The COM port, you need to select the right one for your device. Try unplugging it and plugging it back to make sure you have the right one
- The COM settings (Baud rate, Byte Size, Parity, Stopbits...) have to be exactely as set in your Detla DTB, otherwise it will not work and they will not communicate. Check the [DTB user manual](https://deltaacdrives.com/Delta-DTB-User-Manual.pdf) to see how to check and/or modify those values. 
- The slave number is also depending on how you set it in your DTB. in #716 screenshots it is set as n°2. Most devices are set at n°1 by default. Again check in the user manual to check which value you are set to.
Note: when checking in the usermanual, the registers are often in hexadecimal values, you will need to convert them to decimal to use in artisan. If not sure how to, you can use any [online converter](https://www.rapidtables.com/convert/number/hex-to-decimal.html?x=000D) for that.

As mentionned above, I uset the Fuji PXF, but the logic is the same (values will be different) you can check my posts [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-fuji-pxf-and-test-of-maxwell-mta) and [here](https://www.bigloose.com/v11/index.php/home/archive/artisan-roasting-with-the-pid-using-bt-as-input-fuji-pxf) to have an idea how I did mysetup. Reading the 2 posts may help you to find what is not correct in your setup.

Nico

**Comment by finermax1** [comment]
Hello
Thanks for the reply. I did install the driver.
I saw the device in device manager application. However, the pid doesn’t
communicate with the computer. Ask Delta, got a link to download solo
software. It doesn’t work either. I did the scan, see nothing as it
supposed to pick up the pid.
The ifd6500 has cable to direct connect the pid. Try it. Doesn’t work.Try
the Modbus setting. Rtu 8 n 1 slave 1register 18176 com port 4. Doesn’t
work.ordering a Rs485 to usb direct connecter. Guess it won’t work either.
No Ip router Ethernet in the reply.

Thanks

L

**Reply by Terracotta-6** [reply]
The problem has nothing to do with artisan，Download DTCOM,  configure the temperature controller.
![baa91cad5aba524eae8e4faf5ca03dc](https://github.com/artisan-roaster-scope/artisan/assets/149906183/0f883d0f-7994-46ac-93f6-0275b9a7ece0)

**Comment by finermax1** [comment]
Hello:

Thanks for telling me. I downloaded and tested it. Keep getting the message
of the specified com port doesn't exist.

How to solve this problem? If I could configure the com port in the pid, it
will solve the problem. If I couldn't, that means it specified by itself, I
must find out which com port it is.

How others take on this?

Thank you

L

**Reply by Terracotta-6** [reply]
There is a hardware problem or the temperature controller is not correctly selected, there is no serial interface itself, and there is a problem with the rs485 to usb device.

**Comment by finermax1** [comment]
Hello :

 I try another usb socket. I seems better. The artisan has better reaction.
However, the temp doesn't show either.

thanks

L

**Comment by finermax1** [comment]
Hello:

I changed another usb port. I saw the temp on the software now. The artisan
needs more effort to show temp.

Thanks

L

**Reply by poundy** [reply]
i don't understand all your tests or what you mean by Artisan needs more effort. Unless YOU have all the settings right, no software will work.  Your earlier test with the software that the vendor gave you ("solo software") when that didn't work its likely because you gave it an incorrect com port. The simplest way to figure out what you need to be giving any application is to use Device Manager. You open Device Manager without the PID connected. You then look at what serial ports, if any, are visible on the system already. Then, you insert the PID USB cable and see what port or ports now become visible. If it's only one, easy, that's the one you use. If it's two then it becomes a lottery and you have to try both.  And yes, sometimes the PC hardware can cause the COM number to change depending on which physical port you connect the cable to - just be consistent in what you plug in where. And you can always fall back to Device Manager to find out what port you should be using...

**Comment by finermax1** [comment]
Hello:

Thanks for the confirmation. I successfully to show temp in DTCOM with wire
connected to ifd6500. However, I want to show the temp in artisan with
wireless function.
Do you know how to do that. For example the setting time out delay?

Thanks

L

**Reply by poundy** [reply]
It is clear to me that you, and only you, have this problem. And you are the only one with this combination of equipment. Nobody else is likely to be able to help guide you. 

But the IFD6500 you keep mentioning is WIRED. There's no woreless. Just use the cable?

**Comment by Terracotta-6** [comment]
[DELTA_IA-TC_DTB_I_TSET_20140421.pdf](https://github.com/artisan-roaster-scope/artisan/files/13330682/DELTA_IA-TC_DTB_I_TSET_20140421.pdf)

**Comment by finermax1** [comment]
Hello:
 I am confused too. Somebody did that.
(1) Connect Delta DTA to Artisan (home-barista.com)
<https://www.home-barista.com/roasting/connect-delta-dta-to-artisan-t41643.html>
I guess RS485 is a bluetooth connection that can be used in PC.

Thanks

L

**Reply by Terracotta-6** [reply]
The wireless connection of the temperature control instrument requires the addition of hardware, and there are many hardware options. The discussion here is beyond scope.

**Comment by finermax1** [comment]
Hello:

What equipment is required? I would rather go back to phidget sbc4 to roast
wirelessly.

Thanks

L

**Reply by bigl00z3** [reply]
`The Phidget sbc 4 is not wireless either. you will also need additional hardware. ` <-- not correct, see below edit
Your issue is more in terms of computer connection/configuration. Do you have anybody you know who is good at IT or an engineer? 
They probably can help you if you have them read this thread and help you. Your issue is not related to Artisan but more on how to connect a device to your computer and configure the communication port and protocol values for the RS485/Modbus.

Wired or wireless will then depend on the hardware converter you choose to go from your Delta DTP to your usb.

Nico

Edit:
correcting my text, when looking at the [sbc4 documentation,](https://www.phidgets.com/?prodid=969#Wireless) yes it is wireless

**Comment by finermax1** [comment]
Hello:

 I use the phidget sbc 4 with wireless n adapter now. Can I use them in
cropster to roast wirelessly too?

Thank you

L

**Reply by poundy** [reply]
Why would you expect anyone in the Artisan support community to be able to tell you about using Cropster?
